File: 백준/Silver/9613. GCD 합/GCD 합.c
#include <stdio.h>

int gcd(int a, int b) {
	if (b == 0) return a;
	else return gcd(b, a % b);
}

int main(void) {
	int n, m, arr[100];
	scanf("%d", &n);

	while (n--) {
		scanf("%d", &m);
		for (int i = 0; i < m; i++) {
			scanf("%d", &arr[i]);
		}
		long long int sum = 0;
		for (int i = 0; i < m - 1; i++) {
			for (int j = i + 1; j < m; j++) {
				sum += gcd(arr[i], arr[j]);
			}
		}
		printf("%lld\n", sum);
	}

	return 0;
}

File: 백준/Silver/9613. GCD 합/GCD 합.py
def gcd(a, b):
    while(b > 0):
        a, b = b, a % b
    return a

n = int(input())

while(n > 0):
    num = list(map(int, input().split()))
    m = num[0]
    arr = num[1: m + 1]
    sum = 0
    for i in range(m - 1):
        for j in range(i + 1, m):
            sum += gcd(arr[i], arr[j])
    print(sum)
    
    n -= 1